src/org/usfirst/frc/team2473/robot/subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,176 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package org.usfirst.frc.team2473.robot.subsystems;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+import org.usfirst.frc.team2473.framework.Devices;
+import org.usfirst.frc.team2473.robot.RobotMap;
+
+import com.ctre.phoenix.ErrorCode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+
+public class DriveSubsystem extends Subsystem {
+	
+	private static DriveSubsystem instance;
+	
+	static {
+		instance = new DriveSubsystem();
+	}
+	
+	public static DriveSubsystem getInstance() {
+		return instance;
+	}
+	
+	WPI_TalonSRX backLeft;
+	WPI_TalonSRX backRight;
+	WPI_TalonSRX frontLeft;
+	WPI_TalonSRX frontRight;
+	
+	DifferentialDrive teleopDrive;
+	
+	private HashMap<Double, Double> leftTable = new HashMap<>();
```

Comment:
Might be good to write down what the inputs and outputs of these lookup tables are.